lib/ansible/modules/system/mdadm.py

```diff
@@ -0,0 +1,366 @@
+    description:
+      - Number of spare devices in RAID build.
+    required: true
+    default: 0
+  assume_clean:
+    description:
+      - Assume created array is clean.
+    default: no
+  force:
+    description:
+      - Force raid creation.
+    default: no
+author: Adam Hamsik <haaaad@gmail.com>
+'''
+
+EXAMPLES = '''
+mdadm:
+  state: present
+  level: 1
+  md_device: /dev/md0
+  raid_devices:
+    - /dev/sda
+    - /dev/sdb
+    - /dev/sdc
+  spare_devices: 1
+  force: no
+
+mdadm:
+  state: present
+  level: 5
+  md_device: auto
+  raid_devices:
+    - /dev/sda
+    - /dev/sdb
+
+mdadm:
+  state: examine
+  md_device: /dev/md0
+
+mdadm:
+  state: examine
+  raid_devices:
+    - /dev/sda
+    - /dev/sdb
+
+mdadm:
+  state: scan
+register: list_of_raid_devices_on_system
+'''
+
+RETURN = '''
+#
+'''
+
+# TODO:
+#  1) Add support for starting RAID with some missing disks
+#  3) It seems that mdadm returns different hash if it creates raid
+#       device and when it's already created.
+#  4) Use mdstat python library output
+#
+
+RAID_CONF = {
+    0:{'min_dev': 1, 'redundancy': 0},
+    1:{'min_dev': 2, 'redundancy': 1},
+    5:{'min_dev': 3, 'redundancy': 1},
+    6:{'min_dev': 4, 'redundancy': 2},
+    10:{'min_dev': 4, 'redundancy': 2}
+}
+
+class MdManager(object):
+    '''
+    MDADM raid manager class
+    '''
+    def __init__(self, module, md_device, level, raid_devices, spares):
+        self.module = module
+        self.level = level
+        self.md_device = md_device
+        self.raid_devices = raid_devices
+        self.spares = spares
+
+        self.test_mdadm()
+
+    def test_mdadm(self):
+        ''' Test if mdadm is actuall executable or not '''
+        test_cmd = "mdadm --version"
+
+        (md_rc, del_out, del_err) = self.module.run_command(test_cmd, check_rc=True)
+
+    def test_raid_level_disks_number(self):
+        ''' Check if we have proper number of disks for givel RAID level. '''
+        if not RAID_CONF.get(self.level, None):
+            self.module.fail_json(msg="Unsupported RAID level '%d' used"% self.level, level=self.level)
+
+        min_dev = RAID_CONF[self.level]['min_dev']
+        disk_red = 0 #TODO: intentionaly set to 0 until missing disk support is done RAID_CONF[level]['redundancy']
+
+        # Starting Number of devices in RAID Array can be lower that min_dev - redundancy
+        if (min_dev - disk_red) > len(self.raid_devices):
+            self.module.fail_json(msg="Not enough RAID disks provided '%d' for level '%d', minimum is '%d'"
+                                  % (len(self.raid_devices), self.level, (min_dev - disk_red)),
+                                  level=self.level, raid_devices=self.raid_devices)
+
+        return True
+
+    def test_raid_disks(self, level, md_device):
+        ''' Check if there is a raid superblock already present on given disks '''
+
+        for disk in self.raid_devices:
+            test_cmd = "mdadm --examine --test %s" % disk
```

Comment:
I know this is a small one, but I can imagine % formatting will be decommissioned at some point, I would change this to .format() method if it's not too much of a hassle. 

To back-up this up, ansible is undergoing an effort to be Python3 compatible, as per Python docs the recommendation is to prefer string.format() method against string % formatting operator.  [1][2]

Also keep in mind that % might have undesired effects [3] 

[1] https://docs.python.org/2/library/stdtypes.html#str.format
[2] https://www.python.org/dev/peps/pep-3101/
[3] http://stackoverflow.com/questions/5082452/python-string-formatting-vs-format